src/query/auth_query_handler.hpp

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include <optional>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include "query/frontend/ast/ast.hpp"  // overkill
+#include "query/typed_value.hpp"
+
+namespace memgraph::query {
+
+class AuthQueryHandler {
+ public:
+  AuthQueryHandler() = default;
+  virtual ~AuthQueryHandler() = default;
+
+  AuthQueryHandler(const AuthQueryHandler &) = delete;
+  AuthQueryHandler(AuthQueryHandler &&) = delete;
+  AuthQueryHandler &operator=(const AuthQueryHandler &) = delete;
+  AuthQueryHandler &operator=(AuthQueryHandler &&) = delete;
+
+  /// Return false if the user already exists.
```

Comment:
delete all method comments since we don't know what will the actual implementation look like

Reply:
This is what was there from before (refactor move from elsewhere)
This comment is a contractual obligation, all implementations should match the contract.

Reply:
This is what was there from before (refactor move from elsewhere)
This comment is a contractual obligation, all implementations should match the contract.